cpp/src/dataframe/type_dispatcher.hpp

```diff
@@ -116,6 +116,44 @@ decltype(auto) type_dispatcher(gdf_dtype dtype,
   return return_type();
 }
 
-} // namespace gdf
+
+// Only difference is lack of CUDA_HOST_DEVICE_CALLABLE
+template < class functor_t, 
+           typename... Ts>
+decltype(auto) host_type_dispatcher(gdf_dtype dtype, 
+                                    functor_t f, 
+                                    Ts&&... args)
+{
+  switch(dtype)
+  {
+    // The .template is known as a "template disambiguator" 
+    // See here for more information: https://stackoverflow.com/questions/3786360/confusing-template-error
+    case GDF_INT8:      { return f.template operator()< int8_t >(std::forward<Ts>(args)...); }
+    case GDF_INT16:     { return f.template operator()< int16_t >(std::forward<Ts>(args)...); }
+    case GDF_INT32:     { return f.template operator()< int32_t >(std::forward<Ts>(args)...); }
+    case GDF_INT64:     { return f.template operator()< int64_t >(std::forward<Ts>(args)...); }
+    case GDF_FLOAT32:   { return f.template operator()< float >(std::forward<Ts>(args)...); }
+    case GDF_FLOAT64:   { return f.template operator()< double >(std::forward<Ts>(args)...); }
+    case GDF_DATE32:    { return f.template operator()< date32 >(std::forward<Ts>(args)...); }
+    case GDF_DATE64:    { return f.template operator()< date64 >(std::forward<Ts>(args)...); }
+    case GDF_TIMESTAMP: { return f.template operator()< timestamp >(std::forward<Ts>(args)...); }
+    case GDF_CATEGORY:  { return f.template operator()< category >(std::forward<Ts>(args)...); }
+    case GDF_STRING:    // had to add these to avoid host-compiler warnings that aren't there  
```

Comment:
I had some reports of this same issue coming up for the normal type dispatcher as well. My solution was just to add a `default` case with `assert(false)`. It's not ideal, but in a release build it will silently fail, and on a debug build it will crash the application. See here on my new PR: https://github.com/rapidsai/cudf/pull/436/files#diff-0baddc8d0100d6813482b810e105a77cR113

Reply:
Did you have an opinion on having a separate host dispatcher?